src/rgw/driver/rados/rgw_rados.cc

```diff
@@ -1422,10 +1422,21 @@ int RGWRados::log_list_init(const DoutPrefixProvider *dpp, const string& prefix,
     delete state;
     return r;
   }
-  state->prefix = prefix;
-  state->obit = state->io_ctx.nobjects_begin();
-  *handle = (RGWAccessHandle)state;
-  return 0;
+  try {
+    state->prefix = prefix;
+    state->obit = state->io_ctx.nobjects_begin();
+    *handle = (RGWAccessHandle)state;
+    return 0;
+  } catch (const std::system_error& e) {
+    r = -e.code().value();
+    ldpp_dout(dpp, 10) << "nobjects_begin threw " << e.what()
```

Comment:
is 10 the right log level for this kind of error?

Reply:
i think 10 is a good choice. if this ends up being a fatal error, we should already be logging that at level 0 or 1 further up the call stack

Reply:
will we have more than an error code at that point in the unwind?